compiler/resolution/tuples.cpp

```diff
@@ -584,6 +587,129 @@ static void instantiate_tuple_init(FnSymbol* fn) {
 *                                                                             *
 ************************************** | *************************************/
 
+void addTupleCoercion(AggregateType* fromT,
```

Comment:
This function is too long. How about split it into shorter chunks?

For example, the code handling the five cases can be moved out to a separate function - or two five separate functions.

Also, the USR_FATAL_CONT call at the beginning can fit in two lines instead of four.
The formals can fit in three lines. These will make the function easier to read.